File: problem3.pl
#! /usr/bin/env perl

use strict;
use warnings; 

sub largest_prime_factor {
  my $original = shift;
  my @factors = ();
  my $largestFactor = 0;

  for (my $x = 2; ($x**2) < $original; $x++) {
    if ( $original % $x == 0 ) {

      $factors[0] = $x;
      $factors[1] = $original/$x;

      for my $fact(@factors) {
        my $isPrime = 1;
        for ( my $z = 2; ($z**2) < $fact; $z++) {
          if ( $fact % $z == 0 ) {
            $isPrime = 0;
            last;
          }
        }
        if ( $isPrime && $fact > $largestFactor ) {
          $largestFactor = $fact;
        }
      }
    }
  }
  return $largestFactor;
}
my $factor = largest_prime_factor(600851475143);
warn $factor;

File: problem3.js
#! /usr/bin/node

function largest_prime_factor(num) {

  var original = num
  var factors = [];
  var largestFactor = 0;

  //brute force this
  for (var x = 2; (x*x) < original; x++) {
    if ( original % x == 0 ) {
      factors[0] = x;
      factors[1] = original/x;

      for (var y = 0; y < 2; y++ ) {
        console.log(factors);
        var isPrime = true;
        for ( var z = 2; (z*z) < factors[y]; z++) {
          if ( factors[y] % z == 0 ) {
            isPrime = false;
            break;
          }
        }
        if ( isPrime === true && factors[y] > largestFactor ) {
          console.log(factors[y]);
          largestFactor = factors[y];
        }
      }
    }
  }
  return largestFactor;
}
//var sum = largest_prime_factor(13195);
//29
var factor = largest_prime_factor(600851475143);
console.log(factor);

console.log(600851475143/sum);

